### ISSUE TITLE
screenshot action hangs while pdf works fine

### ISSUE THREAD

**Issue by mikeys** [segmentio/nightmare#809]
Hi guys,

Trying to leverage `nightmare` in order to be able to convert an svg to a png.
For some reason the following gets stuck on the `screenshot` action:

``` node
var Nightmare = require('nightmare')
var nightmare = Nightmare({show: true})
var url = "file:///Users/username/Pictures/whatever.svg"
var userAgent = "Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/38.0.2125.111 Safari/537.36"

nightmare
  .useragent(userAgent)
    .goto(url)
  .wait('svg')
  .evaluate(() => {
    const svgElem = document.querySelector('svg')

    return {
      width: svgElem.scrollWidth,
      height: svgElem.scrollHeight
    }
  })
  .then(function(dimensions) {
    return nightmare
      .viewport(dimensions.width, dimensions.height)
      .screenshot('./screenshot.png')
  })
  .then(function() {
    console.log('done')
    nightmare.end()
  })
  .catch(function(err) {
    console.log('error', err)
  })
```

Logs:

```
  nightmare queuing process start +0ms
  nightmare queueing action "useragent" +2ms
  nightmare queueing action "goto" for file:///Users/username/Pictures/whatever.svg +1ms
  nightmare queueing action "wait" +1ms
  nightmare queueing action "evaluate" +0ms
  nightmare running +1ms
  nightmare:actions .useragent() to Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/38.0.2125.111 Safari/537.36 +360ms
  nightmare:log Navigating: "file:///Users/username/Pictures/whatever.svg",
            headers: [none],
            timeout: 30000 +8ms
  nightmare:log did-start-loading +63ms [{}]
  nightmare:log did-get-response-details +77ms [{},true,"file:///Users/username/Pictures/whatever.svg","file:///Users/username/Pictures/whatever.svg",200,"GET","",{"access-control-allow-origin":["*"]},"mainFrame"]
  nightmare:log dom-ready +91ms [{}]
  nightmare:log did-frame-finish-load +0ms [{},true]
  nightmare:log did-finish-load +1ms [{}]
  nightmare:log did-stop-loading +1ms [{}]
  nightmare:actions .wait() for svg element +1ms
  nightmare:actions .evaluate() fn on the page +6ms
  nightmare queueing action "viewport" +3ms
  nightmare queueing action "screenshot" +0ms
  nightmare running +0ms
  nightmare:actions .viewport() +0ms
  nightmare:actions .screenshot() +29ms
  nightmare:log subscribing to browser window frames +1ms
  nightmare:log altering page to force rendering +0ms
```

using `pdf` instead of `screenshot` does not get stuck. Any idea what may be causing this?

**Comment by daniel3d** [comment]
Check this out https://github.com/segmentio/nightmare/issues/761

**Comment by rosshinkley** [comment]
@daniel3d is right: `.end()` queues the ending action, it still needs to be executed with a `.then()`.

Setting that aside, and this is kind of offtopic: I'm not certain Nightmare is the best tool for converting SVGs to PNGs.  Have you considered using something like [ImageMagick](http://imagemagick.org/script/index.php)?  `convert` will give you _much_ more flexibility for image manipulation.

**Comment by mikeys** [comment]
@rosshinkley unfortunately not for SVGs :)

**Comment by Mr0grog** [comment]
Anyone here experiencing timeouts may want to give the #927 a try. There’s also second approach to solving this over on https://github.com/mr0grog/nightmare/tree/screenshot-with-offscreen-rendering that you might give a try.

**Comment by rosshinkley** [comment]
Considering this fixed with #927.  Closing.
